pilot/pkg/model/service.go

```diff
@@ -386,9 +387,19 @@ func WorkloadInstancesEqual(first, second *WorkloadInstance) bool {
 	if first.Endpoint == nil || second.Endpoint == nil {
 		return first.Endpoint == second.Endpoint
 	}
-	if first.Endpoint.Address != second.Endpoint.Address {
+
+	if len(first.Endpoint.Addresses) == len(second.Endpoint.Addresses) {
+		firtAddrs := first.Endpoint.SortIstioEndpointByAddresses()
```

Comment:
Mutating the resource in an `Equal` function is unexpected and undesired. Plus you can do this more efficiently - see listEqualUnordered in ads.go.

However, again, I strongly encourage us to treat this as an ordered list NOT an unordered one. Or always sort it during creation. We should not be sorting on each access

Reply:
Hi @howardjohn, okay, got it. I create a `IsEqualIstioEndpoint` function for it. And it's okay to treat all the addresses of 
 each Endpoint as an ordered list, then check the equality by using `IsEqualIstioEndpoint`.